core/shared/src/main/scala/chartreuse/Plot.scala

```diff
@@ -50,19 +52,197 @@ final case class Plot[
+
+    val scale = Scale.linear.build(dataBoundingBox, width, height)
+
+    val xTicks = TickMarkCalculator.calculateTickScale(minX, maxX, 12)
+    val yTicks = TickMarkCalculator.calculateTickScale(minY, maxY, 12)
+
+    // Map the Ticks to the screen coordinates
+    val xTicksMapped = Ticks(
+      scale(Point(xTicks.min, 0)).x,
+      scale(Point(xTicks.max, 0)).x,
+      xTicks.size
+    )
+    val yTicksMapped = Ticks(
+      scale(Point(0, yTicks.min)).y,
+      scale(Point(0, yTicks.max)).y,
+      yTicks.size
+    )
+
+    // Convert the Ticks to a sequence of points
+    val xTicksSequence = xTicksToSequence(xTicks, scale)
+    val yTicksSequence = yTicksToSequence(yTicks, scale)
+
+    val allLayers =
       layers
         .map(_.draw(width, height))
         .foldLeft(empty[Alg])(_ on _)
-        .margin(20)
-        .debug(color = Color.black)
-        .margin(5)
 
+    val plotWithXTicks = withXTicks(xTicksSequence, allLayers, yTicksMapped)
+    val plotWithXAndYTicks =
+      withYTicks(yTicksSequence, plotWithXTicks, xTicksMapped)
+    val plotWithTicksAndAxes =
+      withAxes(plotWithXAndYTicks, xTicksMapped, yTicksMapped)
+
+    if (grid) {
+      val plotWithTicksAndAxesAndGrid = withGrid(
+        plotWithTicksAndAxes,
+        xTicksMapped,
+        yTicksMapped,
+        xTicksSequence,
+        yTicksSequence
+      )
+      return withTitles(plotWithTicksAndAxesAndGrid)
+    }
+
+    withTitles(plotWithTicksAndAxes)
+  }
+
+  /** Converts `Ticks` to a list of tuples. The first element is the mapped
+    * coordinate of a tick - to place the tick on a graph. The second one is the
+    * original coordinate - to give the tick a label with its coordinate.
+    */
+  private def xTicksToSequence(
+      xTicks: Ticks,
+      scale: Bijection[Point, Point]
+  ): TicksSequence = {
+    (0 to ((xTicks.max - xTicks.min) / xTicks.size).toInt)
+      .map(i =>
+        (
+          scale(Point(xTicks.min + i * xTicks.size, 0)),
+          Point(xTicks.min + i * xTicks.size, 0)
+        )
+      )
+      .toList
+  }
+
+  /** Converts `Ticks` to a list of tuples. The first element is the mapped
+    * coordinate of a tick - to place the tick on a graph. The second one is the
+    * original coordinate - to give the tick a label with its coordinate.
+    */
+  private def yTicksToSequence(
+      yTicks: Ticks,
+      scale: Bijection[Point, Point]
+  ): TicksSequence = {
+    (0 to ((yTicks.max - yTicks.min) / yTicks.size).toInt)
+      .map(i =>
+        (
+          scale(Point(0, yTicks.min + i * yTicks.size)),
+          Point(0, yTicks.min + i * yTicks.size)
+        )
+      )
+      .toList
+  }
+
+  private def withXTicks(
+      xTicksSequence: TicksSequence,
+      plot: Picture[Alg, Unit],
+      yTicksMapped: Ticks
+  ): Picture[Alg, Unit] = {
+    xTicksSequence
+      .foldLeft(plot)((plot, tick) =>
+        plot
+          .on(
+            OpenPath.empty
+              .moveTo(tick._1.x, yTicksMapped.min - 10)
```

Comment:
I'm a bit not-fan of using these methods on tuple. I'd rather destructure with an irrefutable match like so:

```scala
val (screen, _) = tick
```